src/buffer/out/TextAttribute.cpp

```diff
@@ -358,4 +387,5 @@ void TextAttribute::SetStandardErase() noexcept
 {
     _extendedAttrs = ExtendedAttributes::Normal;
     _wAttrLegacy = 0;
+    _hyperlinkId = 0;

```

Comment:
We're gonna have to figure this one out, because SetStandardErase is called in a few different places... not all of which should reset the active link.

In the case of...

```
\e];;sup\e\\I am a \e[4mUNDERLINED LINK\e[m (only there)\e]8;;\e\\
```

we drop the hyperlink after `LINK` because we SetStandardErase during `\e[m`.

gnome-terminal doesn't do this. There are a few things gnome-terminal _preserves hyperlinks_ during... including scrolling a new line on, clearing, etc. but not DECALN (`\e#8`)

Not sure how we should handle this one. Maybe setting standard erase just _shouldn't(?)_ end the link?

James may have a better-informed understanding here...

Reply:
Hmm... this is an interesting case. Logically I think I would expect hyperlinks to be cleared by any operation using `SetStandardErase` - the problem is just that SGR 0 shouldn't fall into that category, because it's not actually erasing anything (that's my fault for being lazy with the SGR refactor). So I'd probably recommend adding a new method (something like `SetDefaultMetaAttributes`) that SGR 0 can use to reset _extendedAttrs and _wAttrLegacy, and then you can leave `SetStandardErase` also clearing the hyperlinks.

And I need to double check, but I _think_ that is the way VTE works as well. However, I've only tested on the version that's included with Ubuntu LTS, and that's a few years out of date. Ideally we should be matching VTE exactly, since this sequence is their invention, but if it turns out that they're doing something weird, I'd want to double with them check whether that is intended behaviour, and not just a bug, before we blindly copy them.

Reply:
SGR 0 now only resets extended/meta attrs